plugins/linkattacher/src/linkattacherserverimpl.cpp

```diff
@@ -0,0 +1,192 @@
+
+        //Display all the links
+        //yInfo() << "Attach --> Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
+
+        std::size_t lastcolon = candidate_robot_link_name.rfind(":");
+        std::string unscoped_robot_link_name = candidate_robot_link_name.substr(lastcolon+1,std::string::npos);
+        //yInfo() << "Attach --> Unscoped robot link name " << unscoped_robot_link_name;
+
+        if(unscoped_robot_link_name == robot_link_name)
+        {
+            yInfo() << LogPrefix << "Attach: Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
+            gazebo::physics::LinkPtr robot_link = robot_model_links[i];
+            if(!robot_link)
+            {
+                yError() << LogPrefix << "Attach: " << robot_link_name << " is not found";
+                return false;
+            }
+            else yInfo() << LogPrefix << "Attach: " << robot_link->GetName() << " found";
+
+            //This is joint creation
+            gazebo::physics::JointPtr joint;
+
+            #if GAZEBO_MAJOR_VERSION >= 8
+            joint = _world->Physics()->CreateJoint("fixed",object_model);
+            #else
+            joint = _world->GetPhysicsEngine()->CreateJoint("fixed",object_model);
+            #endif
+
+            if(!joint)
+            {
+                yError() << LogPrefix << "Attach: Unable to create joint";
+                return false;
+            }
+
+            std::string joint_name = model_link_name + "_magnet_joint";
+            joint->SetName(joint_name);
+            yInfo() << LogPrefix << "Attach: Magnet joint: " << joint->GetName() << " created";
+
+            joint->SetModel(object_model);
+
+            #if GAZEBO_MAJOR_VERSION >= 8
+            joint->Load(object_link,robot_link,ignition::math::Pose3d());
+            #else
+            joint->Load(object_link,robot_link,gazebo::Math::Pose());
+            #endif
+
+            //Attach(prent_link,child_link)
+            joint->Attach(object_link,robot_link);
+
+            break;
+        }
+    }
+
+    return true;
+}
+
+bool LinkAttacherServerImpl::detachUnscoped(const string& model_name, const std::string& model_link_name)
+{
+    #if GAZEBO_MAJOR_VERSION >= 8
+    gazebo::physics::ModelPtr object_model = _world->ModelByName(model_name);
+    #else
+    gazebo::physics::ModelPtr object_model = _world->GetModel(model_name);
+    #endif
+
+    if(!object_model)
+    {
+        yError() << LogPrefix << "Detach: " << model_name << " does not exist in gazebo";
+        return false;
+    }
+    else yInfo() << LogPrefix << "Detach: " << object_model->GetName() << " found";
+
+    gazebo::physics::LinkPtr object_link = object_model->GetLink(model_link_name);
+    if(!object_link)
+    {
+        yError() << LogPrefix << "Detach: " << model_link_name << " is not found";
+        return false;
+    }
+    else yInfo() << LogPrefix << "Detach: " << object_link->GetName() << " found";
```

Comment:
do not forget this, as well.

Reply:
I will leave it as it is because it gives a proper debug code for the given rpc command

Reply:
the bracketing ;D

Reply:
Done